orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/adapters/RetryableTaskTasklet.groovy

```diff
@@ -72,4 +89,11 @@ class RetryableTaskTasklet extends TaskTasklet {
     }
     return super.doExecuteTask(stage, chunkContext)
   }
+
+  /**
+   * In order to accurately determine whether this task has timed out, any time spent in a paused state must be discounted.
+   */
+  static long determineNow(Clock clock, PausedDetails pausedDetails) {
+    return clock.millis() - ((pausedDetails?.pausedMs ?: 0) as Long)
```

Comment:
Small thing but it feels like this method could be better named or just even encapsulate the whole check in a method rather than getting values back & doing a comparison in the caller.